src/emulator/controller.ts

```diff
@@ -588,6 +592,14 @@ export async function startAll(
       port: eventarcAddr.port,
     });
     await startEmulator(eventarcEmulator);
+
+    const tasksAddr = legacyGetFirstAddr(Emulators.TASKS);
+    const tasksEmulator = new TasksEmulator({
+      host: tasksAddr.host,
+      port: tasksAddr.port,
+    });
+
+    await startEmulator(tasksEmulator);
```

Comment:
Yes, the task queue emulator is run whenever the functions emulator is run.

There is the code here: https://github.com/firebase/firebase-tools/blob/03297d5ccb0b8ebfe8dfd654d748f6b9fc997b48/src/emulator/tasksEmulator.ts#L215

which only starts the listening logic of the task queue emulator once an actual task has been enqueued to avoid doing unnecessary updates while the user isn't doing anything with task queues in their functions code. The express server is still spun up every time the functions emulator starts though.

Reply:
Hmm I wonder if we should instead have Task Queue Emulators be explicitly started, just like the Pub/sub emulator.

@blidd-google thoughts?